src/main/java/uk/gov/hmcts/reform/professionalapi/controller/internal/OrganisationInternalController.java

```diff
@@ -697,6 +703,61 @@ public ResponseEntity<OrganisationEntityResponse> retrieveOrganisationByUserId(
         return organisationService.retrieveOrganisationByUserId(userId);
     }
 
+    @Operation(
+        summary = "Deletes the provided list of dx addresses from the organisation.",
+        description = "**IDAM Roles to access API** : <br> - prd-admin",
+        security = {
+            @SecurityRequirement(name = "ServiceAuthorization"),
+            @SecurityRequirement(name = "Authorization")
+        }
+    )
+    @ApiResponse(
+        responseCode = "204",
+        description = "Successfully deleted the list of provided payment accounts from the organisation.",
+        content = @Content
+    )
+    @ApiResponse(
+        responseCode = "400",
+        description = DEL_ORG_PBA_NOTES_1 + DEL_ORG_PBA_NOTES_2 + DEL_ORG_PBA_NOTES_3
+            + DEL_ORG_PBA_NOTES_4 + DEL_ORG_PBA_NOTES_5,
+        content = @Content
+    )
+    @ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized Error : "
+            + "The requested resource is restricted and requires authentication",
+        content = @Content
+    )
+    @ApiResponse(
+        responseCode = "403",
+        description = "Forbidden Error: "
+            + "Access denied for either invalid permissions or user is pending",
+        content = @Content
+    )
+    @ApiResponse(
+        responseCode = "404",
+        description = "Resource Not Found Error: The Organisation does not exist"
+            + " to delete Payment Accounts from",
+        content = @Content
+    )
+    @ApiResponse(
+        responseCode = "500",
+        description = "Internal Server Error",
+        content = @Content
+    )
+
+    @DeleteMapping(path = "/dxAddress/{orgId}")
+    @ResponseStatus(value = HttpStatus.NO_CONTENT)
+    @Secured({"prd-admin"})
+    public void deleteDxAddressOfOrganisation(
+        @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "deletedxAddressRequest")
+        @Valid @NotNull @RequestBody DxAddressCreationRequest dxAddressRequest,

```

Comment:
The DxAddressCreationRequest class has both dxNumber and dxExchange properties as mandatory. However, for deleting an dx address, only dxNumber is used. Can you rename Request to a more generic name, like DxAddressRequest, as the current name is confusing for delete operations? Additionally, please remove the @NotNull annotation on the dxExchange property. For the old logic, manually handle the null check for the dxExchange field and return an error where necessary.